File: Problem-7/main.c
#include <stdio.h>

// Problem #7
// By listing the first six prime 
// numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

// What is the 10 001st prime number?

int prime_numbers(int value);
int main(void)
{
    int x = 0;
    int flag = 0;
    for (int i = 1; i <= 5000000; i++)
    {
         x = prime_numbers(i);
         if (x != 0)
         {
            flag++;
            if (flag == 10001)
            {
                printf("%d\n",x);
                break;
            }
         }
         
    }
  
    return 0;
}


int prime_numbers(int value)
{

  int n, i, flag = 0;
  n = value;
  // 0 and 1 are not prime numbers
  // change flag to 1 for non-prime number
  if (n == 0 || n == 1)
    flag = 1;

  for (i = 2; i <= n / 2; ++i)
  {
    // if n is divisible by i, then n is not prime
    // change flag to 1 for non-prime number
    if (n % i == 0)
    {
      flag = 1;
      break;
    }
  }

  // flag is 0 for prime numbers
  if (flag == 0)
    return n;
}

File: Problem-7/main.lua

-- Problem #7
-- By listing the first six prime 
-- numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
-- What is the 10 001st prime number?

local x = 0
local fl = 0
function PrimeNumbers(value)

    local n = value
    local flag = 0

    if n == 0 or n == 1 then
        flag = 1;
    end

    for i = 2, n / 2, 1 do
        if n % i == 0 then
            flag = 1
            break
        end
    end

    if flag == 0 then
        return n
    end
end

for i = 1, 5000000, 1 do
    local y = true
    x = PrimeNumbers(i)
    if x == nil  then
    else
        fl = fl + 1
        if fl == 10001 then
            print(x)
            break
        end
    end
end

